extension/js/common/core/pgp-msg.ts

```diff
@@ -211,6 +213,38 @@ export class PgpMsg {
   }
 
   public static encrypt: PgpMsgMethod.Encrypt = async ({ pubkeys, signingPrv, pwd, data, filename, armor, date }) => {
+    // slice(1) filters first key that is always own OpenPGP key
+    // in result if X.509 encryption is selected the e-mail author
+    // cannot decrypt e-mails that they have sent
+    const keyTypes = new Set(pubkeys.slice(1).map(PgpMsg.getKeyType));
+    if (keyTypes.size > 1) {
+      throw new Error('Mixed key types are not allowed: ' + [...keyTypes]);
+    }
+    const keyType = keyTypes.keys().next().value;
+    if (keyType === 'x509') {
+      const p7 = forge.pkcs7.createEnvelopedData();
+
+      for (const pubkey of pubkeys.slice(1)) {
```

Comment:
better filter into two groups, pgp and s/mime, in separate variables, and evaluate those